src/rules/utils/misc.ts

```diff
@@ -16,7 +16,11 @@ import { type ParsedExpectFnCall, isTypeOfJestFnCall } from './parseJestFnCall';
 
 const REPO_URL = 'https://github.com/jest-community/eslint-plugin-jest';
 
-export const createRule = ESLintUtils.RuleCreator(name => {
+export interface JestPluginDocs {
+  requiresTypeChecking?: boolean;
+}
+
+export const createRule = ESLintUtils.RuleCreator<JestPluginDocs>(name => {
```

Comment:
https://typescript-eslint.io/blog/announcing-typescript-eslint-v8-beta/#custom-rule-metadocs-types

In theory there's no hard need in plugins to mention `requiresTypeChecking` anymore... except, in practice `eslint-doc-generator` reads it:

https://github.com/jest-community/eslint-plugin-jest/blob/4cc2a1b680507ff006b5d2b02fa6d262584bb263/.eslint-doc-generatorrc.js#L25